File: practice12/6.cpp
//https://www.nowcoder.com/practice/8f3df50d2b9043208c5eed283d1d4da6?tpId=37&tqId=21228&tPage=1&rp=&ru=/ta/huawei&qru=/ta/huawei/question-ranking
#include<iostream>
#include<string>
#include<math.h>
using namespace std;
int main()
{
	string s;
	char ch;
	int i, k, len, b1, b2, q;
	while (cin >> s)
	{
		len = s.length();
		q = len - 3;
		k = 0;
		for (i = 2; i < len; i++)
		{
			ch = s[i];
			if (ch >= '0'&&ch <= '9'&&ch - '0' < 16)   //处理0-9的字符，不能包含大于等于进制数的字符
			{
				b1 = s[i] - '0';
				k = k + b1 * pow(16, q--);
			}
			if (ch >= 'A'&&ch <= 'Z' && (ch - 'A') < 6)   //处理A-F的字符，不能包含大于等于进制数的字符
			{
				b2 = s[i] - '7';
				k = k + b2 * pow(16, q--);
			}
		}
		cout << k << endl;
	}
	system("pause");
	return 0;
}

File: practice28/6.java
import java.util.*;
//https://www.nowcoder.com/practice/3b4996b5786f4157a3a04a9df81aef08?tpId=98&tqId=33046&tPage=12&rp=12&ru=/ta/2019test&qru=/ta/2019test/question-ranking
public class Main {
    public static void main(String args[]) {
        Scanner scanner=new Scanner(System.in);
        int n=scanner.nextInt();
        int k=scanner.nextInt();
        List<Integer> pt=new ArrayList<>();
        while(k>0){
            pt.add(k%2);
            k=k/2;
        }
        int sum=0;
        int x=pt.size()-1;
        for(int i=pt.size()-1;i>=0;i--){
            sum= (int) (sum+pt.get(i)*Math.pow(n,x--));
        }
        System.out.println(sum);
    }
}